exasol/analytics/audit/audit.py

```diff
@@ -0,0 +1,74 @@
+from inspect import cleandoc
+from typing import List
+
+from exasol.analytics.audit.columns import AuditColumns
+from exasol.analytics.query_handler.query.select import (
+    AuditQuery,
+    ModifyQuery,
+    SelectQueryWithColumnDefinition,
+)
+from exasol.analytics.schema import (
+    Column,
+    SchemaName,
+    TableLikeName,
+    TableLikeNameImpl,
+    decimal_column,
+    timestamp_column,
+    varchar_column,
+)
+
+
+class TableDescription:
+    """
+    This class describes an SQL table by its attributes
+    * table (instance of (TableLikeName)) defining the name and optionally the
+      database schema of the SQL table
+    * columns the names and types of the columns of the SQL table
+
+    The class also offers a property `create` which renders the attributes
+    into an SQL CREATE TABLE statement. The class is tested together with
+    AuditTable in an integration test.
+    """
+
+    def __init__(self, table: TableLikeName, columns: List[Column]):
+        self.table = table
+        self.columns = {c.name.name: c for c in columns}
+
+    @property
+    def create(self):
+        columns = ",\n  ".join(c.for_create for c in self.columns.values())
+        return cleandoc(
+            f"""
+            CREATE TABLE IF NOT EXISTS {self.table.fully_qualified} (
+              {columns}
+            )
+            """
+        )
+
+
+class AuditTable(TableDescription):
+    def __init__(self, db_schema: str):
+        super().__init__(
+            table=TableLikeNameImpl("AUDIT_LOG", SchemaName(db_schema)),
+            columns=AuditColumns.all,
```

Comment:
You cannot define all AuditColumns in this project, because extensions will add their own audit columns. You can only define a subset, that is used by the AuditLogWriter itself and provide a separate list for the StageGraph, which can be added, when a extension uses the StageGraph

Reply:
see next push